### PR TITLE
[EOnline] Add new extractor (Closes #6198)

### LINKED ISSUE #6198: EOnline Videos - UnsupportedError
I tried to download a video from www.eonline.com  but it currently is not supported by youtube-dl. Below are details. Thanks for your help.

```
youtube-dl -v -o "/media/test.%(ext)s" "http://www.eonline.com/videos/237478/check-out-the-craziest-celeb-comments"
[debug] System config: [u'--ffmpeg-location', u'/home/mary/bin/ffmpeg']
[debug] User config: []
[debug] Command-line args: [u'-v', u'-o', u'/media/test.%(ext)s', u'http://www.eonline.com/videos/237478/check-out-the-craziest-celeb-comments']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2015.07.07
[debug] Python version 2.7.9 - Linux-3.19.0-21-generic-x86_64-with-Ubuntu-15.04-vivid
[debug] exe versions: ffmpeg 2.6.git, ffprobe 2.6.git, rtmpdump 2.4
[debug] Proxy map: {}
[generic] check-out-the-craziest-celeb-comments: Requesting header
WARNING: Could not send HEAD request to http://www.eonline.com/videos/237478/check-out-the-craziest-celeb-comments: HTTP Error 405: Method Not Allowed
[generic] check-out-the-craziest-celeb-comments: Downloading webpage
WARNING: Falling back on generic information extractor.
[generic] check-out-the-craziest-celeb-comments: Extracting information
ERROR: Unsupported URL: http://www.eonline.com/videos/237478/check-out-the-craziest-celeb-comments
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/generic.py", line 1092, in _real_extract
    doc = parse_xml(webpage)
  File "/usr/local/bin/youtube-dl/youtube_dl/utils.py", line 1562, in parse_xml
    tree = xml.etree.ElementTree.XML(s.encode('utf-8'), **kwargs)
  File "/usr/lib/python2.7/xml/etree/ElementTree.py", line 1300, in XML
    parser.feed(text)
  File "/usr/lib/python2.7/xml/etree/ElementTree.py", line 1642, in feed
    self._raiseerror(v)
  File "/usr/lib/python2.7/xml/etree/ElementTree.py", line 1506, in _raiseerror
    raise err
ParseError: not well-formed (invalid token): line 141, column 30
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 654, in extract_info
    ie_result = ie.extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 273, in extract
    return self._real_extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/generic.py", line 1663, in _real_extract
    raise UnsupportedError(url)
UnsupportedError: Unsupported URL: http://www.eonline.com/videos/237478/check-out-the-craziest-celeb-comments
```

_UPD by dstftw:_
Geo restricted example URL: `http://www.eonline.com/videos/237478/check-out-the-craziest-celeb-comments`. US or CA IP is required.
Geo unrestricted example URL: `http://uk.eonline.com/videos/237549/ariana-grande-posts-video-apology-for-doughnut-fiasco`

### PR DESCRIPTION
**Pull request by TRox1972**

### CODE REVIEW

**Comment by TRox1972** [comment]
Any comments?

**Comment by TRox1972** [comment]
Test currently fails due to some changes on the site. I'm looking in to it now.

File: youtube_dl/extractor/eonline.py [hunk 1]
```diff
@@ -2,41 +2,46 @@
...
+                release_url = entry['videoSourceUrl']
 
         return {
-            '_type': 'url_transparent', 
+            '_type': 'url_transparent',
             'ie_key': 'ThePlatform',
-            'url': smuggle_url(update_url_query(release_url, {'mbr': 'true', 'switch': 'http'}),
-                {'force_smil_url': True}),
+            'url': smuggle_url(update_url_query(release_url, {'mbr': True, 'switch': 'http'}), {'force_smil_url': True}),
+            'id': video_id,
```
Comment (TRox1972) [inline youtube_dl/extractor/eonline.py]: Should I return the `video_id` as the `id`, or leave it up to ThePlatform; they are not the same.

**Comment by TRox1972** [comment]
OK. Fixed the extractor.

**Comment by TRox1972** [comment]
@dstftw @yan12125 Any comments?

**Comment by TRox1972** [comment]
Fixed in 433af6ad3002424ecb316e23946722d54010dbe1

**Comment by tv21** [comment]
I think the URL format for full episodes may have changed.  Please see my comment on https://github.com/rg3/youtube-dl/issues/6198

**Pull request closed without merging**